qiskit_experiments/characterization/spectroscopy.py

```diff
@@ -0,0 +1,408 @@
+        The spectroscopy pulse has the following parameters:
+        - amp: The amplitude of the pulse must be between 0 and 1, the default is 0.1.
+        - duration: The duration of the spectroscopy pulse in samples, the default is 1000 samples.
+        - sigma: The standard deviation of the pulse, the default is duration / 4.
+        - width: The width of the flat-top in the pulse, the default is 0, i.e. a Gaussian.
+
+        Args:
+            qubit: The qubit on which to run spectroscopy.
+            frequencies: The frequencies to scan in the experiment.
+            unit: The unit in which the user specifies the frequencies. Can be one
+                of 'Hz', 'kHz', 'MHz', 'GHz'. Internally, all frequencies will be converted
+                to 'Hz'.
+            absolute: Boolean to specify if the frequencies are absolute or relative to the
+                qubit frequency in the backend.
+
+        Raises:
+            QiskitError: if there are less than three frequency shifts or if the unit is not known.
+
+        """
+        if len(frequencies) < 3:
+            raise QiskitError("Spectroscopy requires at least three frequencies.")
+
+        if unit not in self.__units__:
+            raise QiskitError(f"Unsupported unit: {unit}.")
+
+        self._frequencies = [freq * self.__units__[unit] for freq in frequencies]
+        self._absolute = absolute
+
+        super().__init__([qubit], circuit_options=("amp", "duration", "sigma", "width"))
+
+    def circuits(self, backend: Optional["Backend"] = None, **circuit_options):
+        """Create the circuit for the spectroscopy experiment.
+
+        The circuits are based on a GaussianSquare pulse and a frequency_shift instruction
+        encapsulated in a gate.
+
+        Args:
+            backend: A backend object.
+            circuit_options: Key word arguments to run the circuits. The circuit options are
+                - amp: The amplitude of the GaussianSquare pulse, defaults to 0.1.
+                - duration: The duration of the GaussianSquare pulse, defaults to 1024 samples.
+                - sigma: The standard deviation of the GaussianSquare pulse, defaults to one
+                    fifth of the duration.
+                - width: The width of the flat top in the GaussianSquare pulse, defaults to 0.
+
+        Returns:
+            circuits: The circuits that will run the spectroscopy experiment.
+
+        Raises:
+            QiskitError: If relative frequencies are used but no backend was given.
+        """
+        if not backend and not self._absolute:
+            raise QiskitError("Cannot run spectroscopy relative to qubit without a backend.")
+
+        amp = circuit_options.get("amp", 0.1)
+        duration = circuit_options.get("duration", 1024)
+        sigma = circuit_options.get("sigma", duration / 4)
+        width = circuit_options.get("width", 0)
+
+        drive = pulse.DriveChannel(self.physical_qubits[0])
```

Comment:
This is not necessary. You can use `pulse.drive_channel` in the builder (you need to give backend).
```python
with pulse.build(backend=backend, name=...) as sched:
   play(..., pulse.drive_channel(self.physical_qubits[0]))
```
This takes channel-qubit mapping from the backend and thus we don't need to assume qubit-ch index matching here.

Reply:
See cb8e5e2